src/main/resources/templates/src/main/java/MainVerticle.java.ftl

```diff
@@ -1,12 +1,20 @@
 package ${packageName};
 
 import io.vertx.core.AbstractVerticle;
+<#if vertxVersion?index_of("3") != 0>
+import io.vertx.core.Promise;
+<#else>
 import io.vertx.core.Future;
+</#if>
 
 public class MainVerticle extends AbstractVerticle {
 
   @Override
+  <#if vertxVersion?index_of("3") != 0>
```

Comment:
This test would be more robust with [`starts_with`](https://freemarker.apache.org/docs/ref_builtins_string.html#ref_builtin_starts_with)